app/demo-loop/LDemoLauncher.hh

```diff
@@ -185,6 +217,53 @@ CELER_FUNCTION void AlongAndPostStepLauncher<M>::operator()(ThreadId tid) const
             crossed_boundary          = geo_step.boundary;
         }
 
+        // Sample the multiple scattering
+        if (phys.msc_ppid())
+        {
+            const celeritas::detail::UrbanMscNativeRef& urban_data
+                = params_.physics.hardwired.urban_data;
+
+            if (step > urban_data.params.geom_limit
+                && particle.energy() > urban_data.params.energy_limit)
+            {
+                using UrbanMscScatter = celeritas::detail::UrbanMscScatter;
+
+                msc_step_result.geom_path = step;
+                UrbanMscScatter msc_scatter(urban_data,
+                                            particle,
+                                            geo.dir(),
+                                            phys,
+                                            mat.material_view(),
+                                            msc_step_result);
+                auto            msc_result = msc_scatter(rng);
+                step                       = msc_result.step_length;
+
+                // Update direction and position
+                geo.set_dir(msc_result.direction);
+
+                real_type rho = norm(msc_result.displacement);
+                if (rho > urban_data.params.geom_limit)
+                {
+                    // Update position
+                    celeritas::Real3 new_pos = geo.pos();
+                    real_type safety = (1 - urban_data.params.safety_tol)
+                                       * geo.find_safety(geo.pos());
+                    if (rho <= safety)
+                    {
+                        axpy(real_type(1.0), msc_result.displacement, &new_pos);
+                    }
+                    else if (safety > urban_data.params.geom_limit)
+                    {
+                        axpy(safety / rho, msc_result.displacement, &new_pos);
+                    }
```

Comment:
I think most of this can stay inside the MSC scatter result where you'd put it, right? The only difference I intended from your previous version is that instead of calling `move_internal` and `set_dir` inside UrbanMscScatter, you call them from *outside*. You could still pass the `geo` in, or *always* calculate safety (probably too expensive?), but just have inside the UrbanMSCScatter:
```suggestion
                    real_type safety = (1 - urban_data.params.safety_tol)  * safety;
                    real_type multiplier = 0;
                    if (rho <= safety)
                    {
                        multiplier = 1;
                    }
                    else if (safety > urban_data.params.geom_limit)
                    {
                        multiplier = safety / rho;
                    }
                    for (auto i : range(3))
                    {
                        result.displacement[i] *= multiplier;
                    }
```